src/oracle/oracle_utils.cairo

```diff
@@ -96,10 +107,11 @@ fn validate_block_number_within_range(
 /// # Returns
 /// True if block_number is in range, false else.
 fn is_block_number_within_range(
-    min_oracle_block_numbers: Array<u128>, max_oracle_block_numbers: Array<u128>, block_number: u128
+    min_oracle_block_numbers: Span<u64>, max_oracle_block_numbers: Span<u64>, block_number: u64
 ) -> bool {
-    // TODO
-    true
+    let lower_bound = min_oracle_block_numbers.max().unwrap();
+    let upper_bound = max_oracle_block_numbers.min().unwrap();
+    lower_bound <= block_number && block_number <= upper_bound
```

Comment:
Update function signature and implement logic to check if block number is within the range of oracle block numbers provided